src/hooks/useWebSocket.ts

```diff
@@ -0,0 +1,116 @@
+'use client';
+import { SOCKET } from '@/constants/websocket';
+import { ChatMessage } from '@/types';
+import * as StompJS from '@stomp/stompjs';
+import { useRef, useState } from 'react';
+
+export function useWebSocket() {
+  const BASE_WEBSOCKET_URL = process.env.NEXT_PUBLIC_BASE_WEBSOCKET_URL;
+  const client = useRef<StompJS.Client | null>(null);
+  const [, setSubscription] = useState<StompJS.StompSubscription | null>(null);
+  const [chatMessages, setChatMessages] = useState<ChatMessage[]>([]);
+
+  const connect = (roomId: string) => {
+    client.current = new StompJS.Client({
+      brokerURL: BASE_WEBSOCKET_URL,
```

Comment:
P5; 문득 든 생각인데 이런 방어코드는 있는게 좋을까요 없는게 좋을까요? 의견 부탁드립니다! @kim-hyunjoo 
```suggestion
  const connect = (roomId: string) => {
    if (!BASE_WEBSOCKET_URL) {
      throw new Error('NEXT_PUBLIC_BASE_WEBSOCKET_URL is not defined.');
    }
    client.current = new StompJS.Client({
      brokerURL: BASE_WEBSOCKET_URL,
```

Reply:
오... 처음엔 상수로 명확히 정의되어 있을텐데 굳이 필요한가? 싶었는데
저희가 은근히 많이 겪는 오류가 새로운 개발 환경에서 .env 파일이 없는 상황에서 실행했을 때, 그 원인을 파악 못할 때가 많았던 것 같아요.
그래서 환경 변수 관련 방어 코드를 넣어주면 디버깅 시 바로 원인을 파악하기 쉽기 때문에 있으면 좋을 것 같네요!

Reply:
개발 환경에서 env파일이 있는지 없는지 알 수 있어서 좋은 것 같아요. 추가하고 머지할게요!